dbms/src/Storages/KVStore/MultiRaft/ApplySnapshot.cpp

```diff
@@ -225,6 +265,8 @@ void KVStore::onSnapshot(
                     static_assert(std::is_same_v<RegionPtrWrap, RegionPtrWithBlock>);
                     // Call `deleteRange` to delete data for range
                     dm_storage->deleteRange(new_key_range, context.getSettingsRef());
+                    // We must flush the deletion to the disk here, because we only flush new range when persisting this region later.
+                    dm_storage->flushCache(context, new_key_range, /*try_until_succeed*/ true);
```

Comment:
I think line 297 also does the `flushCache` job. What is the different between these two lines?

https://github.com/pingcap/tiflash/blob/14007a151c5a838e9faaa5bde4558d7d0fe80213/dbms/src/Storages/KVStore/MultiRaft/ApplySnapshot.cpp#L290-L298

Reply:
The stage 1 just makes sure the stale data is cleaned and persisted.
The stage 2 is to decode some data in new region and then trying to flush the decoded data.

What will be improved if we save the flushCache call here? I checked the code, seems it would save a `prepare` call and a `commit` call, each of which will result in a page storage IO?